test/Libraries/Microsoft.Extensions.AotCompatibility.TestApp/Microsoft.Extensions.AotCompatibility.TestApp.csproj

```diff
@@ -0,0 +1,50 @@
+<Project>
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>$(LatestTargetFramework)</TargetFramework>
+    <!-- clear TargetFrameworks because it is set in ..\Directory.Build.props -->
+    <TargetFrameworks></TargetFrameworks>
+    <PublishAot>true</PublishAot>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Remove the ReferenceTrimmer analyzer. We explicitly want to reference every assembly so the AOT compiler sees it. -->
+    <PackageReference Remove="ReferenceTrimmer" />
+
+    <LibraryProjects Include="$(RepoRoot)\src\Libraries\**\*.csproj" />
+    <!-- Azure.AI.Inference produces many warnings -->
```

Comment:
Did we open issues on these repos so that we can track them fixing the warnings on their side? If so, we can probably link those here too.

Also, is there q way to configure the trimmer so that it ignores warnings on certain assemblies or baseline the existing warnings? Mostly asking so we can ensure that dotnet/extensions side is not adding more warnings on top of the external ones.

Reply:
OpenAI has an issue that I will update the code to link to: https://github.com/openai/openai-dotnet/issues/20.

I've opened [[FEATURE REQ] Resolve Native AOT Warnings in Azure.AI.Inference (Azure/azure-sdk-for-net#47069)](https://github.com/Azure/azure-sdk-for-net/issues/47069) for Azure.AI.Inference.

Reply:
> Also, is there q way to configure the trimmer so that it ignores warnings on certain assemblies or baseline the existing warnings? Mostly asking so we can ensure that dotnet/extensions side is not adding more warnings on top of the external ones.

There is not an easy way to do this outside of dotnet/runtime and dotnet/aspnetcore. The feature we use there isn't an officially supported feature.

Note that only the 5 libraries can be affected. For any other library, or new libraries, they are expecting 0 warnings.

Reply:
Yup, I was mostly talking about those 5 libraries, so that we don't add more  warnings in those unintentionally. Anyway, if not really possible without some hacking, then no need to block this for that.

Reply:
Also - the Roslyn analyzers are still enabled, so that will catch a decent amount of problems as well.